provider/datagen/src/source.rs

```diff
@@ -77,6 +82,16 @@ impl SourceData {
         })
     }
 
+    /// Adds TZDB data to this `DataSource`. The root should point to the directory
+    /// `cldr-{version}-json-{full, modern}.zip` directory or ZIP file (see
```

Comment:
Where do these files actually come from? I couldn't find them in the IANA package.

We want to have `with_tzdb_for_tag` and `with_tzdb_latest` methods that download the files without the user having to do anything.

Reply:
These files come from downloading the time-zone database, building the binaries using the `Makefile` and then running the Zone Info Compiler `zic` to generate the `TZif` files for all the time zones. 

I would _love_ to have automatic download options for this in the future, but this process is going to be much harder to automate. 

It would leave Windows users out of luck, unless they had `make` installed already. 

For now, I think it's best to land a manual solution and work on an automated download PR later, since it will be unfortunately more involved than simply downloading CLDR JSON, for example.

Reply:
> Not just windows, we don't want to rely on certain tools (like make) being available on any platform.

Yes, that is what I figured. It is a shame. 

There should not be any issues surrounding platform-specific details with the files themselves.

Just for the sake of clarity, I want to outline the details of how these files get generated.

---

> 1) **Download the IANA database (https://www.iana.org/time-zones)**

The database consists of many different human-readable files, in their own unique format (i.e. not JSON, YAML etc.), but there are a number of C binaries (buildable through `make`) that can act on this human-readable data.

---

> 2) **Build the binaries using `make`**

---

> 3) **Run the Zone Info Compiler binary (`zic`) to generate Time Zone Information Format (TZif) files**

The Zone Info Compiler parses the human-readable files in the IANA database and turns them into a binary format which follows a specification.

https://datatracker.ietf.org/doc/rfc8536/

Earlier this year, I wrote a parser for the TZif files which exists in ICU4X under `utils/tzif`. 

But TZif is a common file format for this data. Linux distributions such as Ubuntu already have TZif files that get shipped with the operating system. (Note also that Ubuntu also ships a binary of `zic` as well as the pre-compiled TZif files).

The reasoning behind ICU4X using TZif was that users could have a number of different sources that they might want to generate their ICU4X data. 

* They could download the IANA database directly and generate TZif files from that. 
* They could pull TZif files from their operating system if their operating system contains them natively. 
* They could get TZif files from presumably some other source.

Reply:
Definitely makes sense for us to use the standardised compiled format. What doesn't really make sense is IANA not publishing it.